compiler/rustc_const_eval/src/check_consts/check.rs

```diff
@@ -548,6 +548,10 @@ impl<'tcx> Visitor<'tcx> for Checker<'_, 'tcx> {
                 }
             }
 
+            Rvalue::RawPtr(RawPtrKind::FakeForPtrMetadata, _) => {
+                // Do nothing.
```

Comment:
Uh, this is dangerous. What is the justification for allowing *all* these raw borrows in const? It seems to be this relies on a non-local invariant about where and how `FakeForPtrMetadata` can occur. I am not a fan if such invariants; I'd prefer the const checker to be correct *at least* for all MIR that passes the MIR validator.